### PR TITLE
Floor trap

### PR DESCRIPTION
**Pull request by Nimfar11**
## About the PR

Several types of floor traps that are virtually invisible to the eye. If stepped on, they explode, hit with an EMP or summon one of the aggressive mobs. For mappers.

## Why / Balance

The variety for expeditions, wrecks, or anywhere else increases the difficulty and adds surprises to the game. There are traps with xenos, carp and space animals spawning.

Allows you to avoid filling areas with a bunch of mobs, mixing existing mobs with those who will spawn when the player is around.

## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

ADMIN Nimfar11

- add: Adds three types of virtually invisible floor traps for mappers. Explosion, EMP strike and teleport of mobs.

### CODE REVIEW

**Comment by pigeonpeas** [comment]
i feel there should be some sort of indicator that there is a trap, something small so that it still isn't noticeable. Maybe like a cracked tile or something?

**Comment by Nimfar11** [comment]
> i feel there should be some sort of indicator that there is a trap, something small so that it still isn't noticeable. Maybe like a cracked tile or something?

изображение

**Comment by pigeonpeas** [comment]
> > i feel there should be some sort of indicator that there is a trap, something small so that it still isn't noticeable. Maybe like a cracked tile or something?
> You can make it even more visible.

oooh i see!! sorry for bothering, its pretty hard to see the indicators in the video haha

File: Resources/Prototypes/floor_trap.yml [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+  parent: CollideFloorTrapSpawn
+  id: SpawnFloorTrapCarp
+  name: spawn floor trap
+  suffix: Carp
+  components:
+    - type: TriggerOnCollide
+      fixtureID: floortrap
+    - type: SpawnOnTrigger
+      proto: MobSpawnFloorTrapCarp
+    - type: DeleteOnTrigger
```
Comment (metalgearsloth) [inline Resources/Prototypes/floor_trap.yml:98]: Why doesn't it just spawn the carp directly? Seems like a waste of an entire entity to just relay this.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/floor_trap.yml (hunk 1 above)
Comment (Nimfar11) [reply Resources/Prototypes/floor_trap.yml:98]: This is done for several reasons:
1. It's prettier.
2. it doesn't happen instantly, and the player has time to realise and react to the appearance of the mob. For this reason, I didn't add the space spider and cobra spawn, which have poisons and can be extremely dangerous by suddenly appearing near the player.
3. The effect itself gives the understanding that the player has entered the trap, it is an indicator of the event. Than if a mobe suddenly appeared near the player, which is more like adminspawn.

Comment (metalgearsloth) [reply Resources/Prototypes/floor_trap.yml]: If you want a timer implement a timer into it, don't spawn entities just to work around it.

**Review by metalgearsloth — APPROVED** [review]

**Pull request merged**